### ISSUE TITLE
List of DIN Fasteners

### ISSUE THREAD

**Issue by tobiasfalk** [shaise/FreeCAD_FastenersWB#321]
Still need sorting but here is a list of DIN standards and there coresponding ISO standard with chages.
The list is base on [Fuller](https://fullerfasteners.com/tech/din-iso-en-crossover-chart/)

- [ ] DIN 1 Unhardened taper pins **; ISO 2339, Nominal length redefined (with round end)**
- [ ] DIN 7 Parallel pins of unhardened steel and austenitic stainless steel **; ISO 2338, Nominal length redefined (with round end) form A+C deleted**
- [X] DIN 84 Slotted cheese head screws – Product grade A **; ISO 1207, Modifications of dimensions < 2,5; dimension of slots changed; partly nominal lengths deleted; strength 8.8 deleted**
- [ ] DIN 85 Slotted pan head screws – Product grade A **; ISO 1580, Thread included; dimensions of head and slot changed; strength 8.8 deleted**
- [ ] DIN 94 Slotted set screws with long dog point **; ISO 1234, No modifications of dimensions**
- [X] DIN 96 Slotted round head wood screws

- [ ] DIN 125-1 Product grade A washers with a hardness up to 250 HV, designed for use with hexagon head bolts and nuts **; ISO 7089, ISO 7090, Without chamfer, up to nominal diameter 64**
- [ ] DIN 125-2 Product grade A washers with a hardness from 300 HV, designed for use with hexagon head bolts and nuts **; ISO 7089, ISO 7090, Without chamfer, up to nominal diameter 64**
- [ ] DIN 126 Product grade C washers, designed for use with hexagon head bolts and nuts **; ISO 7091, Nominal sizes changed**

- [X] DIN 315 Fasteners - Wing nuts - Rounded wings 

- [ ] DIN 417 Slotted set screws with long dog point **; ISO 7435, No modifications of dimensions**
- [ ] DIN 427 Slotted headless screws with shank **; ISO 2342**
- [ ] DIN 433-1 Product grade A washers with a hardness up to 250 HV, designed for use with cheese head screws **; ISO 7092, Pyramid diamond hardness 140 HV cancelled; rating changed**
- [ ] DIN 433-2 Product grade A washers with a hardness up to 300 HV, designed for use with cheese head screws **; ISO 7092, Nominal sizes changed**
- [ ] DIN 439-1 Hexagon thin nuts with M 1,6 to M 10 threads **; ISO 4036, Correlation of dimensions with DIN 439-1, except for the width across flat of the nut M10**
- [ ] DIN 439-2 Hexagon thin nuts with M 1,6 to M 52 and M 8×1 to M 52×3 threads **; ISO 4035, ISO 8675, In tables 1 and 2 abbreviations changed and some specifications (thread dimensions) changed, Standard divided by regular type screw threads and fine threads; width across flats according to ISO 272 changed; nominal size range changed; fine thread see ISO 8675; regular type screw thread see ISO 4035**
- [ ] DIN 440 Plain washers – Extra large series, product grade C **; ISO 7094, Nominal sizes changed, nominal size 24 size “a” redefined to the maximum, square-end hole Form R omitted; (standard DIN 440 still valid)**
- [X] DIN 471 Retaining rings for shafts - Normal type and heavy type
- [X] DIN 472 Retaining rings for bores - Normal type and heavy type
- [X] DIN 478 Square head bolts with collar

- [X] DIN 508-1 Roofing and cladding products from metal sheet - Specification for self-supporting products of steel, aluminium or stainless steel sheet - Part 1: Steel **; t-Nut exists in FSData but not selectible in FreeCAD**
- [X] DIN 508-2 Roofing and cladding products from metal sheet - Specification for self-supporting products of steel, aluminium or stainless steel sheet - Part 2: Aluminium **; t-Nut exists in FSData but not selectible in FreeCAD**
- [X] DIN 508-3 Roofing and cladding products from metal sheet - Specification for self-supporting products of steel, aluminium or stainless steel sheet - Part 3: Stainless steel **; t-Nut exists in FSData but not selectible in FreeCAD**
- [ ] DIN 551 Slotted set screws with flat point **; ISO 4766, No modifications of dimensions**
- [ ] DIN 553 Slotted set screws with cone point **; ISO 7434, No modifications of dimensions**
- [ ] DIN 555 Hexagon nuts; Metric Thread, Type g **; ISO 4034, Width across flats according to DIN ISO 272 included; nominal diameter reduced**
- [ ] DIN 558 Hexagon screws, Thread Approximately to Head, Metric Thread, Type g **; ISO 4018, Nominal size extended; width across flats according to DIN ISO 272 included**
- [X] DIN 571 Hexagon head wood screws

- [ ] DIN 601 Hexagon Bolts without Hexagon Nut – with Hexagon Nut, Type g **; ISO 4016, Nominal size extended; width across flats according to DIN ISO 272 included**
- [X] DIN 603 Mushroom head square neck bolts **; ISO 8677**

- [ ] DIN 912 Hexagon socket head cap screws **; ISO 4762, Nominal size range changed; fine thread deleted (ISO 21269)**
- [ ] DIN 913 Hexagon socket set screws with flat point **; ISO 4026, Nominal sizes M1.4, M1.8, M14, M18 and M22 omitted**
- [ ] DIN 914 Hexagon socket set screws with cone point **; ISO 4027, Nominal sizes M1.4, M1.8, M14, M18 and M22 omitted**
- [ ] DIN 915 Hexagon socket set screws with dog point **; ISO 4028, Nominal sizes M1.4, M1.8, M14, M18 and M22 omitted**
- [ ] DIN 916 Hexagon socket set screws with cup point **; ISO 4029**
- [X] DIN 917 Hexagon cap nuts, low type
- [X] DIN 928 Square weld nuts
- [X] DIN 929 Hexagon weld nuts
- [ ] DIN 931-1 Hexagon head bolts with shank; M 1,6 to M 39 screw threads, Product grades A and B **; ISO 4014, Nominal size extended; width across flats according to DIN ISO 272 included**
- [X] DIN 933 Hexagon head screws – Product grades A and B **; ISO 4017, Nominal size extended; width across flats according to DIN ISO 272 included**
- [X] DIN 934 Hexagon nuts; Metric screw threads; Product grades A and B **; ISO 4032, ISO 8673, Nominal diameter changed, height of the nut changed partially; nuts with fine thread specified in DIN EN ISO 8673; nuts with regular type screw thread specified in DIN EN ISO 4032; width across flats according to DIN ISO 272 included**
- [ ] DIN 935-1 Hexagon slotted and castle nuts - Part 1: Metric coarse and fine pitch thread, product grades A and B **; only coarse pitch is implemented, the fine pitch is missing**
- [ ] DIN 935-3 Hexagon slotted nuts - Part 3: Metric coarse pitch thread, product grade C
- [ ] DIN 960 Hexagon bolts, Metric fine thread, Types m and mg **; ISO 8765, Nominal size extended; width across flats according to DIN ISO 272 included**
- [X] DIN 961 M 8×1 to M 100×4 hexagon head bolts with fine pitch thread **; ISO 8676, Nominal size extended; width across flats according to DIN ISO 272 included**
- [ ] DIN 963 Slotted countersunk (flat) head screws (Countersunk heads according to ISO) **; ISO 2009, Nominal size changed; head dimension changed; thread length changed; strength 8.8 cancelled**
- [ ] DIN 964 Slotted raised countersunk (oval) head screws (Countersunk heads according to ISO) **; ISO 2010, Nominal size changed; head dimension changed; thread length changed; strength 8.8 cancelled**
- [ ] DIN 965 Cross recessed countersunk (flat) head screws (Countersunk heads according to ISO) **; ISO 7046-1, Nominal size changed; head dimension changed; thread length changed; strength 8.8 included (see Part 2); penetration depth of cross slots changed**
- [ ] DIN 966 Cross recessed countersunk (oval) head screws (Countersunk heads according to ISO) **; ISO 7047, Nominal size changed; head dimension changed; thread length changed; strength 8.8 omitted; penetration depth of cross slots changed**
- [X] DIN 967 Cross recessed pan head screws with collar - Product grade A
- [ ] DIN 980 All-metal prevailing torque type hexagon nuts **; ISO 7042, Height of nut changed; width across flat according to ISO 272 included; and ISO 10513, Height of nut changed; width across flat according to ISO 272 included**
- [ ] DIN 982 Prevailing torque type hexagon nuts with non-metallic insert **; ISO 7040, ISO 10512, Height of nut changed; width across flat according to ISO 272 included**
- [ ] DIN 985 Prevailing torque type hexagon thin nuts with non-metallic insert **; ISO 10511, Height of nut changed; width across flat according to ISO 272 included**

- [ ] DIN 1433 Clevis pins without head **; ISO 2340, Some other nominal lengths defined**
- [ ] DIN 1434 Clevis pins with head **; ISO 2341, Only one head dimension. Some other nominal lengths defined**
- [ ] DIN 1435 Clevis pins with head **; ISO 2341, Only one head dimension. Some other nominal lengths defined**
- [ ] DIN 1436 Clevis pins with head **; ISO 2341, Only one head dimension. Some other nominal lengths defined**
- [ ] DIN 1440 Plain washers for clevis pins; product grade A **; ISO 8738, Nominal sizes changed**
- [ ] DIN 1443 Clevis pins without head **; ISO 2340**
- [ ] DIN 1444 Clevis pins with head **; ISO 2341, Some other nominal lengths defined**
- [ ] DIN 1470 Grooved pins – Fulllength parallel grooved, with pilot **; ISO 8739, Nominal lengths redefined. Shear rate increased.**
- [ ] DIN 1471 Grooved pins – Fulllength taper grooved **; ISO 8744, Nominal lengths redefined. Shear rate increased.**
- [ ] DIN 1472 Grooved pins – Half-length taper grooved **; ISO 8745, Nominal lengths redefined. Shear rate increased.**
- [ ] DIN 1473 Grooved pins – Full-length parallel grooved, with chamfer **; ISO 8740, Nominal lengths redefined. Shear rate increased.**
- [ ] DIN 1474 Grooved pins – Half-length reverse-taper grooved **; ISO 8741, Nominal lengths redefined. Shear rate increased.**
- [ ] DIN 1475 Grooved pins – One-third-length centre grooved **; ISO 8742, Nominal lengths redefined. Shear rate increased.**
- [ ] DIN 1476 Grooved pins with round head **; ISO 8746, Chamfer redefined**
- [ ] DIN 1477 Grooved pins with countersunk head **; ISO 8747, Chamfer redefined**
- [ ] DIN 1481 Spring-type straight pins – Slotted, heavy duty **; ISO 8752, Specification “non-jamming” included; chamfer redefined; partial change of dimensions**

- [X] DIN 1587 Hexagon cap nuts, high type

- [X] DIN 6319 Spherical washers, conical seats
- [ ] DIN 6325 Parallel pins, Hardened; Tolerance Zone m6 **; ISO 8734, Some other nominal lengths defined**
- [X] DIN 6330 Hexagon nuts with a height of 1,5 d 
- [X] DIN 6331 Hexagon collar nuts with a height of 1,5 d 
- [X] DIN 6340 Washers for clamping devices

- [X] DIN 6799 Retaining washers for shafts 

- [X] DIN 6912 Hexagon socket head cap screws with reduced loadability - Low head, with centre 
- [ ] DIN 6914 High-strength hexagon head bolts with large widths across flats for structural steel bolting **; ISO 7412**
- [ ] DIN 6915 High-strength hexagon nuts with large widths across flats for structural steel bolting **; ISO**
- [ ] DIN 6916 Washers for high tensile bolting in steel structures **; ISO 7416**
- [ ] DIN 6921 Hexagon flange bolts **; ISO 8102, Dimensions partially changed; width across flats according to ISO 272; strength 12.9 omitted**
- [ ] DIN 6923 Hexagon nuts with flange **; ISO 4161, Fine thread omitted to be included in a separate standard later; width across flat M10 changed**
- [ ] DIN 6924 Prevailing torque type hexagon nuts with non-metallic insert **; ISO 7040, ISO 10512, Height of nut changed; width across flat according to ISO 272 included**
- [ ] DIN 6925 Prevailing torque type all-metal hexagon nuts **; ISO 7042, ISO 10513, Height of nut changed; width across flat according to ISO 272 included**
- [ ] DIN 6926 Prevailing torque type hexagon nuts with flange (with non-metallic insert), with metric fine pitch thread **; ISO 12125, Width across flat M10 changed**
- [ ] DIN 6927 Prevailing torque type – All-metal hexagon nuts with flange **; ISO 7044, Width across flat M10 changed**

- [ ] DIN 7343 Spring-type straight pins – Coiled, standard duty **; ISO 8750, Range of nominal diameters and lengths extended. Shear range changed.**
- [ ] DIN 7344 Spring-type straight pins – Coiled, heavy duty **; ISO 8748, Range of nominal diameters and lengths extended. Shear range changed.**
- [ ] DIN 7346 Spring-type straight pins, slotted, light duty **; ISO 3337, Specification “non-jamming” included; chamfer redefined; partial change of dimensions**

- [ ] DIN 7500-1 Thread rolling screws for ISO metric thread – Part 1: Technical specifications for case hardened and tempered screws **; ISO, Normative revision, values for case-hardening depth and hole size revised, torque values reduced**
- [ ] DIN 7504 Cross recessed countersunk head drilling screws with tapping screw thread **; ISO 15480, ISO 15481, ISO 15482, ISO 15483 and ISO 10666, thread ST 3.9 omitted; hexagon head with collar and slot (Form L) omitted**

- [X] DIN 7967 Self Locking Counter Nuts
- [ ] DIN 7976 Slotted set screws with cup point **; ISO 1479, No modifications of dimensions**
- [ ] DIN 7977 Unhardened taper pins with external thread **; ISO 8737, Technical delivery conditions partially changed**
- [ ] DIN 7978 Unhardened taper pins with internal thread **; ISO 8736, Dimensions partially changed**
- [ ] DIN 7979 Parallel pins with internal thread, of unhardened and austenitic stainless steel **; ISO 8733, ISO 8735, Dimensions partially changed**
- [ ] DIN 7981 Cross recessed pan head tapping screws **; ISO 7049, Marginal change of head dimensions. Caution with automatic feed and connection systems**
- [ ] DIN 7982 Countersunk (flat) head tapping screws with cross recess **; ISO 7050, Head dimensions and countersunk angles changed**
- [ ] DIN 7983 Raised countersunk (oval) head; tapping screws with cross recess **; ISO 7051, Head dimensions and countersunk angles changed**
- [X] DIN 7984 Hexagon socket head cap screws with low head with reduced loadability
- [ ] DIN 7985 Recessed raised cheese (fillister) head screws **; ISO 7045, Nominal size changed; head dimension changed; thread length changed; penetration depth of cross slots changed**
- [ ] DIN 7991 Hexagon socket countersunk head screws **; ISO 10462**
- [X] DIN 7996 Cross recessed round head wood screws

- [ ] DIN 9021 Washers; outside diameter ˜ 3 x bolt diameter **; ISO 7093-1, ISO 7093-2, Dimensions partially changed**

**Comment by hasecilu** [comment]
Fasteners with no changes in dimensions could be implemented as alias and it's very easy to maintain and for those which have extra nominal sizes a new feature to extend another fastener type may be needed to avoid duplicating common sizes.
Isn't the DIN 508 T-Slot nut working for you?

**Comment by tobiasfalk** [comment]
> Fasteners with no changes in dimensions could be implemented as alias and it's very easy to maintain and for those which have extra nominal sizes a new feature to extend another fastener type may be needed to avoid duplicating common sizes. 

Yes sounds good, but checking if it really is the same still needs to be done.
I did not confirm everything that is written as change

**Comment by tobiasfalk** [comment]
>  Isn't the DIN 508 T-Slot nut working for you?

Yes I do not see it in FreeCAD it self, the same is with "DIN 1624" thats why I closed the issue direktly #320

**Comment by hasecilu** [comment]
It's on its own group "TSlot", it won't appear in the list in other nuts
![image](https://github.com/shaise/FreeCAD_FastenersWB/assets/53124818/642fd886-19d5-4bfd-bbf3-8d1872d9cbc8)

**Comment by tobiasfalk** [comment]
Ok, had to remove and re-install it, and now it appears

**Comment by hasecilu** [comment]
With #353 you can mark 1143, 1144, 1151, 1152, 1160, 464, 465, 653

**Comment by ToniTen** [comment]
DIN 25201 wedge lock fasteners are not included in this list. I requested them on #400

**Comment by tobiasfalk** [comment]
@ToniTen do you mean

- [] DIN 25201-1 Konstruktionsrichtlinie für Schienenfahrzeuge und deren Komponenten - Schraubenverbindungen - Teil 1: Einteilung, Kategorien der Schraubenverbindungen
- [] DIN 25201-2 Konstruktionsrichtlinie für Schienenfahrzeuge und deren Komponenten - Schraubenverbindungen - Teil 2: Konstruktion - Maschinenbauliche Anwendungen
- [] DIN 25201-3 Konstruktionsrichtlinie für Schienenfahrzeuge und deren Komponenten - Schraubenverbindungen - Teil 3: Konstruktion - elektrische Anwendungen
- [] DIN 25201-4 Konstruktionsrichtlinie für Schienenfahrzeuge und deren Komponenten - Schraubenverbindungen - Teil 4: Sichern von Schraubenverbindungen
- [] DIN 25201-5 Konstruktionsrichtlinie für Schienenfahrzeuge und deren Komponenten - Schraubenverbindungen - Teil 5: Korrosionsschutz
- [] DIN 25201-6 Konstruktionsrichtlinie für Schienenfahrzeuge und deren Komponenten - Schraubenverbindungen - Teil 6: Anschlussmaße
- [] DIN 25201-7 Konstruktionsrichtlinie für Schienenfahrzeuge und deren Komponenten - Schraubenverbindungen - Teil 7: Montage

I think that part 6 has the dimensions, but I am not sure size I do not have access to those

**Comment by Rahix** [comment]
ISO 4762 / DIN 912 is already present :)

**Comment by tobiasfalk** [comment]
> ISO 4762 / DIN 912 is already present :)

 Yes and no, DIN 912 has M1,4, M18, M22, M27, M33, M42, M48, M56, M64, M72, M80, M90 and M100

In #286 it is checked of but in my opinion for DIN 912 the missing sizes need to be added to a DIN 912 type.
